marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/CallBatcherImpl.java

```diff
@@ -457,7 +465,16 @@ private void initialize() {
             logger.warn("threadCount should be 1 or greater -- setting threadCount to number of hosts: " + threadCount);
         }
         if (initialized.getAndSet(true)) return;
-// TODO
+        
```

Comment:
The thread pool should be initialized before the for loop starts submitting tasks to execute on the thread pool.

In other words, consider moving the test and for loop after setting the started flag.